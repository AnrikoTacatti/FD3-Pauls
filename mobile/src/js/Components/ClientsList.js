import React from 'react';
import ClientRow from './ClientRow';
import EditProductEl from './EditClient';
import NewProductEl from './NewClient';
import InfoProductEl from './InfoClient';
import { voteEvents } from '../events';
/*

/*
class ИмяКомпонента extends React.Component {

static propTypes = {
  ... // типы props
};

static defaultProps = {
  ... // начальное значение props
};

constructor(props) {
  super(props);
  ...

  this.state = {
    ... // начальное значение state может быть указано здесь, оно может зависеть от props
  };

}

state = {
  ... // начальное значение state может быть указано здесь, оно может зависеть от this.props
};

render() {
  ...
}

}*/


export default class ProductList extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      strclients: props.clients,
      selectid: null,
      editid: null,
      newproduct: {},
      filderror: { name: null, balans: null },
      newForm: null,
      clientsfilter: true
    }


  }

  setEditClients = (data, index) => {
    this.setState((prevState) => {
      debugger;
      let newprevState = [...prevState.strclients];
      let i = newprevState.findIndex(element => element.id === index);
      newprevState[i] = data;
      return { strclients: newprevState }
    });
    this.setState({ selectid: data.id });
    this.setState({ editid: null });
  }

  setEditid = (data) => {
    this.setState({ editid: data });
    this.setState({ selectid: null });
    this.setState({ newForm: null });
  }

  setSelectid = (data) => {
    this.setState({ selectid: data });
    this.setState({ editid: null });
    this.setState({ newForm: false });
  }

  setfildError = (data) => {
    this.setState({ filderror: data });
  }

  setNewproduct = (data) => {
    this.setState({ newproduct: data });
  }

  setNew = (data) => {
    this.setState({ newForm: data });
  }

  clientsFilter = (data) => {
    this.setState({ clientsfilter: data });
  }



  New = (e) => {
    if (e !== undefined) e.stopPropagation();

    this.setState({ newForm: true });
    this.setState({ selectid: null });
    this.setState({ editid: null });
    this.setState({ filderror: { name: null, balans: null } });
  }

  Censel = () => {
    this.setState({ editid: null });
    this.setState({ newForm: null });
    this.setState({ newproduct: {} });
  }

  Delete = (productid) => {
    let prod = this.state.strclients.filter((val) => {
      return val.id != productid
    });
    this.setState({ strclients: prod });
    this.setState({ editid: null });
  }

  setNewClients = (newproduct) => {
    this.setState((prevState) => {
      let newprevState = [...prevState.strclients, newproduct];
      return { strclients: newprevState }

    });
    this.setState({ newForm: null });
    this.setState({ selectid: newproduct.id });
  }


  componentDidMount = () => {
    voteEvents.addListener('EsetNewProduct', this.setNewClients);
    voteEvents.addListener('EsetEditProduct', this.setEditClients);
    voteEvents.addListener('EsetEditid', this.setEditid);
    voteEvents.addListener('EDelete', this.Delete);
    voteEvents.addListener('ECensel', this.Censel);
    voteEvents.addListener('EsetSelectid', this.setSelectid);

  };

  componentWillUnmount = () => {
    voteEvents.removeListener('EsetNewProduct', this.setNewClients);
    voteEvents.removeListener('EsetEditProduct', this.setEditClients);
    voteEvents.removeListener('EsetEditid', this.setEditid);
    voteEvents.removeListener('EDelete', this.Delete);
    voteEvents.removeListener('ECensel', this.Censel);
    voteEvents.removeListener('EsetSelectid', this.setSelectid);
  };



  render() {
    console.log("render ProductList");
    return (
      <React.Fragment>
        <table border="0" width="100%" cellPadding="5">
          <caption>{this.props.name}</caption>
          <tbody>
            <tr>
              <th><input type="button" value="Все" onClick={this.clientsFilter.bind(null, true)} />
                <input type="button" value="Активные" onClick={this.clientsFilter.bind(null, "active")} />
                <input type="button" value="Заблокированые" onClick={this.clientsFilter.bind(null, "blocked")} />
              </th>
            </tr>
            <tr>
              <th>{this.props.colname.surname}</th>
              <th>{this.props.colname.name}</th>
              <th>{this.props.colname.patronymic}</th>
              <th>{this.props.colname.balans}</th>
              <th>{this.props.colname.status}</th>
              <th>редактировать</th>
              <th>удалить</th>
            </tr>
            {


              this.state.strclients.map((el, index) => {
                if (el.status === this.state.clientsfilter || this.state.clientsfilter === true) {
                  return <ClientRow el={el} key={el.id} selectid={el.id === this.state.selectid && this.state.selectid}
                  />;
                }
              })
            }
          </tbody>
        </table>
        <br />
        <input type="button" value="New Product" onClick={this.New} disabled={Object.keys(this.state.newproduct).length > 0} />

        {console.log("id---" + this.state.editid)}
        {

          this.state.editid !== null && this.state.selectid === null &&
          <EditProductEl key={this.state.editid} editid={this.state.editid} strclientsedit={this.state.strclients.find(element => element.id === this.state.editid)} filderror={this.state.filderror} newproduct={this.state.newproduct}
          />
        }
        {
          this.state.newForm === true && this.state.selectid === null &&
          <NewProductEl key={this.state.editid} filderror={this.state.filderror} newproduct={this.state.newproduct}
            newproductid={this.state.strclients.reduce((acc, curr) => acc.id > curr.id ? acc : curr).id + 1} newproductnumber={this.state.strclients.length}
          />
        }
        {
          this.state.selectid !== null &&
          <InfoProductEl key={this.state.editid} product={this.state.strclients.find(element => element.id === this.state.selectid)} selectid={this.state.selectid} />
        }
      </React.Fragment >
    );
  }
};




/*Синтаксис модулей ES6*/



/*
Синтаксис модулей ES6

export { имя1, имя2, ... };

Экспортировать из данного модуля указанные сущности (классы, переменные, функции...).


export default имя;

Экспортировать «по умолчанию» из данного модуля указанную сущность.


export let имя=значение;
export function имя(аргументы) { тело }
export class имя { описание }

Описать и экспортировать из данного модуля переменную, функцию, класс соответственно.


import { имя1, имя2, ... } from "путь/имямодуля";
import { имя1, имя2, ... } from "./имямодуля";
import { имя1, имя2, ... } from "имямодуля";

Получить экспортированные другим модулем сущности.
Во втором варианте — модуль находится в этой же папке.
В третьем варианте — модуль встроенный.


import { имя as новоеимя } from "путь/имямодуля";

Получить экспортированную другим модулем сущность под другим именем.


import имя from "путь/имямодуля";

Получить сущность, экспортированную «по умолчанию» другим модулем.

*/



