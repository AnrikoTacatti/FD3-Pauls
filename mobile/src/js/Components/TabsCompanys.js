
import React from 'react';
import ClientsList from './ClientsList.js';
import '../css/style.css';
const mtsClients = require('../data/mts.json');
const veclonClients = require('../data/velcom.json');

const tableName = "Таблица ";
const tableColName = {
    surname: "Фамилия",
    name: "Имя",
    patronymic: "Отчество",
    balans: "Баланс",
    status: "Статус"
};

export default class TabsCompanys extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            company: veclonClients,
            companykey: "velcom"

        }
        this.setCompany = this.setCompany.bind(this);
    }


    setCompany(data) {
        debugger;
        switch (data) {
            case "velcom":
                this.setState({ company: veclonClients });
                this.setState({ companykey: data });
                break;
            case "mts":
                this.setState({ company: mtsClients });
                this.setState({ companykey: data });
                break;
            default:
                console.log("Sorry, we are out of " + data + ".");
        }
    }


    render() {
        console.log("render TabsCompanys");
        return (
            <div>
                <input type="button" value="Velcom" onClick={this.setCompany.bind(null, "velcom")} />
                <input type="button" value="MTS" onClick={this.setCompany.bind(null, "mts")} />
                <ClientsList key={this.state.companykey} name={tableName} colname={tableColName} clients={this.state.company} />
            </div>
        );
    }
};

