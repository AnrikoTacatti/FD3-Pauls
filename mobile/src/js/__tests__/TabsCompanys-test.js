"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';

import TabsCompanys from '../components/TabsCompanys';
import ClientsList from '../components/ClientsList';


describe('NEWProduct', () => {

  test('работа NEWProductBooton', () => {
    // создаём тестовую версию компонента
    const component = renderer.create(
      <TabsCompanys />
    );
    // найдём в вёрстке компонента саму кнопку
    const buttonNewProduct = component.root.find(el => {
      /*if (el.props.type == 'button' && el.props.value == "New Product") {
        console.log(el.props);
      }
      console.log("==============");*/
      return el.props.type == 'button' && el.props.value == "New Product"
    });
    // и "нажмём" на неё
    buttonNewProduct.props.onClick();
    /* const form = component.root.find(el => {
       /* if (el => el.props.className == "new-client") {
          console.log(el.children);
        }*/
    /*return el.props.className == "new-client"
  });*/
    /* console.log(component.root.find(el => el.props.className === "new-client"));*/
    expect(component.root.find(el => el.props.className === "new-client")).toBeTruthy();
  });

  test('работа NEWProductBooton input', () => {
    const component = renderer.create(
      <TabsCompanys />
    );
    const buttonNewProduct = component.root.find(el => {
      return el.props.type == 'button' && el.props.value == "New Product"
    });
    // и "нажмём" на неё
    buttonNewProduct.props.onClick();
    const form = component.root.find(el => {
      return el.props.className == "new-client"
    });

    expect(form.findAll(
      (el) => el.type == 'input'
        && el.props.type == 'text'
    ).length).toEqual(5);
  });




  test('работа NEWProduct', () => {

    const tableName = "Таблица ";
    const tableColName = {
      surname: "Фамилия",
      name: "Имя",
      patronymic: "Отчество",
      balans: "Баланс",
      status: "Статус"
    };
    const veclonClients = require('../data/velcom.json');

    const component2 = renderer.create(
      <ClientsList key="velcom" name={tableName} colname={tableColName} clients={veclonClients} />
    );

    component2.getInstance().setNewClients({
      "id": 5,
      "surname": "Иванов blocked velcom 5",
      "name": "Иван 5",
      "patronymic": "Иванович 5",
      "balans": 200,
      "status": "blocked"
    });

    expect(component2.getInstance().state.strclients.length).toEqual(5);


  });

});


describe('EditProduct', () => {

  test('работа EditProduct', () => {

    const tableName = "Таблица ";
    const tableColName = {
      surname: "Фамилия",
      name: "Имя",
      patronymic: "Отчество",
      balans: "Баланс",
      status: "Статус"
    };
    const veclonClients = require('../data/velcom.json');

    const component2 = renderer.create(
      <ClientsList key="velcom" name={tableName} colname={tableColName} clients={veclonClients} />
    );
    let client2 = {
      "id": 2,
      "surname": "Иванов blocked velcom 2 test",
      "name": "Иван 2 test",
    };
    component2.getInstance().setEditClients(client2, 2);

    expect(() => {
      let client = component2.getInstance().state.strclients.find(element => element.id == client2.id);
      return client.surname == client2.surname && client.name == client2.name
    }).toBeTruthy();
  });

});

describe('Filter Product', () => {

  test('работа DeleteProduct', () => {

    const tableName = "Таблица ";
    const tableColName = {
      surname: "Фамилия",
      name: "Имя",
      patronymic: "Отчество",
      balans: "Баланс",
      status: "Статус"
    };
    const veclonClients = require('../data/velcom.json');

    const component2 = renderer.create(
      <ClientsList key="velcom" name={tableName} colname={tableColName} clients={veclonClients} />
    );

    component2.getInstance().Delete(3);
    expect(() => {
      let client = component2.getInstance().state.strclients.find(element => element.id == 3);
      return client === undefined
    }).toBeTruthy();
  });

});

describe('Filter Product Product', () => {

  test('работа FilterProduct 	active', () => {

    const tableName = "Таблица ";
    const tableColName = {
      surname: "Фамилия",
      name: "Имя",
      patronymic: "Отчество",
      balans: "Баланс",
      status: "Статус"
    };
    const veclonClients = require('../data/velcom.json');

    const component = renderer.create(
      <ClientsList key="velcom" name={tableName} colname={tableColName} clients={veclonClients} />
    );
    console.log(component.getInstance().state.strclients.find(element => element.status == "active"));
    component.getInstance().clientsFilter("active");
    const active = component.root.findAll((el) => {
      return el.type === 'tr' && el.props.className === "ClientRow";
    });
    expect(active.length).toEqual(stateactive);
  });
  test('работа FilterProduct blocked', () => {

    const tableName = "Таблица ";
    const tableColName = {
      surname: "Фамилия",
      name: "Имя",
      patronymic: "Отчество",
      balans: "Баланс",
      status: "Статус"
    };
    const veclonClients = require('../data/velcom.json');

    const component = renderer.create(
      <ClientsList key="velcom" name={tableName} colname={tableColName} clients={veclonClients} />
    );
    let stateblocked = component.getInstance().state.strclients.filter(element => element.status == "blocked").length;
    component.getInstance().clientsFilter("blocked");
    const active = component.root.findAll((el) => {
      return el.type === 'tr' && el.props.className === "ClientRow";
    });
    expect(active.length).toEqual(stateblocked);
  });
});




//группирует связанные по логике тесты в один блок
/*describe('my beverage', () => {
  // создаём тестовую версию компонента
  const component = ReactTestUtils.renderIntoDocument(
    <TabsCompanys />
  );



  test('form new', () => {
    component.findRenderedDOMComponentWithClass(component, "new-client");
  });

  test('form new input', () => {
    const form = component.findRenderedDOMComponentWithClass(component, "new-client");
    //выводит в консоль
    //console.log(form.childElementCount);
    // expect - утвеждение и его метод toBe
    expect(form.childElementCount).toBe(7);

  });
});*/

