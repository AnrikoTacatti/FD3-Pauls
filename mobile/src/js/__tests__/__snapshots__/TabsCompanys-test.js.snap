// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  <table
    border="0"
    cellPadding="5"
    width="100%"
  >
    <caption>
      Таблица 
    </caption>
    <tbody>
      <tr>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Все"
          />
          <input
            onClick={[Function]}
            type="button"
            value="Активные"
          />
          <input
            onClick={[Function]}
            type="button"
            value="Заблокированые"
          />
        </th>
      </tr>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          редактировать
        </th>
        <th>
          удалить
        </th>
      </tr>
      <tr
        className="ClientRow"
        onClick={[Function]}
      >
        <td>
           
          Иванов 1 velcom
        </td>
        <td>
          Иван 1
        </td>
        <td>
          Иванович 1
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="удалить"
          />
        </td>
      </tr>
      <tr
        className="ClientRow"
        onClick={[Function]}
      >
        <td>
           
          Иванов 2 velcom
        </td>
        <td>
          Иван 2
        </td>
        <td>
          Иванович 2
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="удалить"
          />
        </td>
      </tr>
      <tr
        className="ClientRow"
        onClick={[Function]}
      >
        <td>
           
          Иванов 3 velcom
        </td>
        <td>
          Иван 3
        </td>
        <td>
          Иванович 3
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="удалить"
          />
        </td>
      </tr>
      <tr
        className="ClientRow"
        onClick={[Function]}
      >
        <td>
           
          Иванов blocked velcom
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>,
  <br />,
  <input
    disabled={false}
    onClick={[Function]}
    type="button"
    value="New Product"
  />,
]
`;

exports[`Filter Product Product работа FilterProduct 	active Snapshot 1`] = `
Array [
  <table
    border="0"
    cellPadding="5"
    width="100%"
  >
    <caption>
      Таблица 
    </caption>
    <tbody>
      <tr>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Все"
          />
          <input
            onClick={[Function]}
            type="button"
            value="Активные"
          />
          <input
            onClick={[Function]}
            type="button"
            value="Заблокированые"
          />
        </th>
      </tr>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          редактировать
        </th>
        <th>
          удалить
        </th>
      </tr>
      <tr
        className="ClientRow"
        onClick={[Function]}
      >
        <td>
           
          Иванов 1 velcom
        </td>
        <td>
          Иван 1
        </td>
        <td>
          Иванович 1
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="удалить"
          />
        </td>
      </tr>
      <tr
        className="ClientRow"
        onClick={[Function]}
      >
        <td>
           
          Иванов 2 velcom
        </td>
        <td>
          Иван 2
        </td>
        <td>
          Иванович 2
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="удалить"
          />
        </td>
      </tr>
      <tr
        className="ClientRow"
        onClick={[Function]}
      >
        <td>
           
          Иванов 3 velcom
        </td>
        <td>
          Иван 3
        </td>
        <td>
          Иванович 3
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          active
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>,
  <br />,
  <input
    disabled={false}
    onClick={[Function]}
    type="button"
    value="New Product"
  />,
]
`;

exports[`Filter Product Product работа FilterProduct 	blocked  Snapshot 1`] = `
Array [
  <table
    border="0"
    cellPadding="5"
    width="100%"
  >
    <caption>
      Таблица 
    </caption>
    <tbody>
      <tr>
        <th>
          <input
            onClick={[Function]}
            type="button"
            value="Все"
          />
          <input
            onClick={[Function]}
            type="button"
            value="Активные"
          />
          <input
            onClick={[Function]}
            type="button"
            value="Заблокированые"
          />
        </th>
      </tr>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          редактировать
        </th>
        <th>
          удалить
        </th>
      </tr>
      <tr
        className="ClientRow"
        onClick={[Function]}
      >
        <td>
           
          Иванов blocked velcom
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="blocked"
        >
          blocked
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="редактировать"
          />
        </td>
        <td>
          <input
            onClick={[Function]}
            type="button"
            value="удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>,
  <br />,
  <input
    disabled={false}
    onClick={[Function]}
    type="button"
    value="New Product"
  />,
]
`;
